x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/EsRelation.java

```diff
@@ -35,30 +38,54 @@ public class EsRelation extends LeafPlan {
         EsRelation::readFrom
     );
 
-    private final EsIndex index;
+    private final String indexName;
+    private final IndexMode indexMode;
+    private final Map<String, IndexMode> indexNameWithModes;
     private final List<Attribute> attrs;
     private final boolean frozen;
```

Comment:
frozen is no longer needed - since you're refactoring the classes, please remove this field from the Es classes.